app/channel.go

```diff
@@ -1234,11 +1229,7 @@ func (a *App) GetPublicChannelsForTeam(teamId string, offset int, limit int) (*m
 }
 
 func (a *App) GetChannelMember(channelId string, userId string) (*model.ChannelMember, *model.AppError) {
-	result := <-a.Srv.Store.Channel().GetMember(channelId, userId)
-	if result.Err != nil {
-		return nil, result.Err
-	}
-	return result.Data.(*model.ChannelMember), nil
+	return a.Srv.Store.Channel().GetMember(channelId, userId)
```

Comment:
Remove wrapper?

Reply:
I don't see removing the wrappers unless they are not used outside the app layer. The app layer encapsulates the business logic, and from the API layer perspective there no such thing called store, there is an App that provide stuff, if we remove the wrapper, we remove the isolation between the API layer and the store layer.

Reply:
I agree with @jespino, I think we need the wrappers for the isolation

Reply:
I am not suggesting we should remove the isolation. However I would like to see the app layer use the store more directly rather than using these wrappers which end up being incapable of their original jobs because they get used everywhere.

But this is out of the scope of this PR.

Reply:
Yes, totally agree with that, we can try to avoid to use this kind of wrappers internally.